### DISCUSSION TITLE
KeePassXC shows no ssh-agen running, even if its running

### DISCUSSION THREAD

**Discussion by molnarti** [keepassxreboot/keepassxc#10216] [Q&A]
KeePassXC always shows there is no ssh agent running, but i can clearly see the agent is on and the SSH_AUTH_SOCK has the correct value 
![image](https://github.com/keepassxreboot/keepassxc/assets/25829862/af5b7aaf-126b-433b-b3a5-cadd91bb2c02)

what could be wrong here?

**Comment by droidmonkey** [comment]
Use the full path, no env vars. Also make sure keepassxc is not running as a snap.

**Comment by ccoenen** [comment]
After upgrading from Fedora 42 to Fedora 43, I now get this same thing, even though SSH-Agent is running. I have KeePassXC in my automatically starting programs, and it seems to be started at a time, where the `SSH_AUTH_SOCK` env var does not exist _yet_.

For me, closing KeePassXC (fully terminating it) and then opening it again is a remedy I can use for now. But at some point I want to find out why this broke for me inthe first place.

**Reply by tomstefanski** [reply]
hey @ccoenen , did you manage to resolve the problem? I've got exactly the same situation after upgrading fedora to v.43.

**Reply by ccoenen** [reply]
Sadly, I did not. I took KeePassXC out of my autostart, and I am currently starting it manually after logon. This, apparently, is late enough so that the env var exists and everything works normally.

**Reply by tomstefanski** [reply]
I think I found a solution (not sure if this is the best method but it seems to be working just fine) -

assuming keepassxc is already taken out of autostart,
- create keepassxc.service in ~/.config/systemd/user/ with the following contents:

```
[Unit]
Description=KeePassXC started after GCR SSH agent
After=gcr-ssh-agent.socket graphical-session.target
Wants=gcr-ssh-agent.socket

[Service]
ExecStart=/usr/bin/keepassxc
Restart=no
Environment=SSH_AUTH_SOCK=/run/user/%U/gcr/ssh

[Install]
WantedBy=graphical-session.target
```

- enable it through 
`systemctl --user enable keepassxc.service` 

- reboot your system to make a test

I think I don't really want to spend too much time on reading what the latest version of Gnome messed up in terms of ssh agent/gcr agent/whatever, but clearly the autostart isn't the right tool [anymore] to guarantee ssh-agent is working when keepassxc wants to use it.
